[TFECO-12319] API DOCS For Registry Tagging and Visibility (#2682)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/api-docs/private-registry/manage-component-versions.mdx

[202]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

[204]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

[400]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

[401]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401

[403]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403

[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404

[409]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409

[412]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412

[422]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

[429]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

[500]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500

[503]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503

[JSON API document]: /terraform/cloud-docs/api-docs#json-api-documents

[JSON API error object]: https://jsonapi.org/format/#error-objects

# Manage component versions API reference

Use the Stack component configuration version management API endpoints to list, add, and update tags for component configuration versions in your organization's registry.

> ## List all tags for a component configuration version

Reviewer: ```suggestion
## List tags
```
We're already on the component versions reference page, we probably don't need to repeat that part in the headings.  Your call.

> Use this endpoint to list all tags assigned to a registry component version.

Reviewer: ```suggestion
Call the following endpoint to list all tags assigned to a registry component version.
```

> `GET /api/v2/registry-component-versions/:component_version-id/tag-bindings`

Reviewer 2: nit: change `:component_version-id` ->`:component_version_id` here and below in the parameters

| Parameter | Description |
| :---- | :---- |
| `:component_version-id` | The specific component configuration version id whose tags you want to list. |

| Status | Response | Reason |
| :---- | :---- | :---- |

> | [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | [JSON API document](/terraform/cloud-docs/api-docs#json-api-documents) | The request was successful. |

Reviewer: ```suggestion
| [200][] | [JSON API document][] | The request was successful. |
```
For all of these links, you can use the reference-style syntax because the URLs are already defined at the beginning of the page.

| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The component configuration version does not exist. |

### Sample request

```shell
curl \
  --request GET \
  --header "Authorization: Bearer $TOKEN" \
  https://app.terraform.io//api/v2/registry-component-versions/compver-JRtoeJbP5R4Fv4qn/tag-bindings
```

### Sample response

```json
{
  "data": [
    {
      "id": "tb-XHJG1sCBejBGrbtS",
      "type": "tag-bindings",
      "attributes": {
        "key": "environment",
        "value": "staging",
        "created-at": "2026-07-15T20:42:14.882Z"
      }
    },
    {
      "id": "tb-k36uoJvKYzTWS1KG",
      "type": "tag-bindings",
      "attributes": {
        "key": "environment",
        "value": "dev",
        "created-at": "2026-07-15T20:42:14.887Z"
      }
    }
  ],
  "links": {
    "target": "/api/v2/registry-components/comp-Xyow7YbAyBDMkax1/versions/0.0.4"
  }
}
```

## Add or update tags for a component configuration version

Use this endpoint to add or update tags for a registry component version in your organization's registry.

> -> **Note:** Removing tags from a component configuration version is made via PATCH requests, where the request body will contain only the tag keys and values that are currently needed.

Reviewer: ```suggestion
## Add or update tags 

Send a `PATCH` request to the following endpoint to add, update, or remove tags for a registry component version in your organization's registry.

The request replaces the existing tags with tag keys and values specified in the request body. To remove specific tags, specify only the tag keys and values you want to keep. A request body with empty `key` and `value` attributes  removes all tags.
```
Correct?

`PATCH /api/v2/registry-components/:component_id/versions/:component_version`

| Parameter | Description |
| :---- | :---- |
| `:component_id` | The id of the registry component whose version you want to tag. |
| `:component_version` | The component configuration version you want to add or update a tag for. |

| Status | Response | Reason |
| :---- | :---- | :---- |
| [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | [JSON API document](/terraform/cloud-docs/api-docs#json-api-documents) | Successfully added or updated the tag binding. |
| [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Request body cannot be parsed. |
| [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) | [JSON API error object](http://jsonapi.org/format/#error-objects) | User not authorized. |
| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The component configuration version does not exist. |

> | [422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Unprocessable content - tag key or value have a wrong format. |

Reviewer: ```suggestion
| [422][] | [JSON API error object][] | HCP Terraform can't process the content because the tag key or value format is incorrect. |
```

### Request body

> This PATCH endpoint requires a JSON object with the following properties as a request payload.

Reviewer: ```suggestion
`PATCH` requests require a JSON object with the following properties as a request payload.
```
